skipLists.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+type Node struct {
+	value int
+	level int     // Store the level of each node.
+	next  []*Node // This allows each node in the skip list to have multiple pointers to other nodes
+} // at different levels, supporting the layered structure of the skip list.
+
+type skipList struct {
+	head         *Node
+	maxNoOfLevel int
+}
+
+func coinToss(maxNoOfLevel int) int {
+	level := 0
+	for level < maxNoOfLevel && rand.Intn(2) == 1 {
+		level++
+	}
+	return level
+}
+
+func newSkipList(maxNoOfLevel int) *skipList {
+	head := &Node{
+		value: 0,
+		level: maxNoOfLevel,
+		next:  make([]*Node, maxNoOfLevel),
+	}
+
+	return &skipList{
+		head:         head,
+		maxNoOfLevel: maxNoOfLevel, // Set the maximum number of levels.
+	}
+}
+
+func (sl *skipList) insert(value int) {
+	newLevel := coinToss(sl.maxNoOfLevel)
+
+	newNode := &Node{
+		value: value,
+		level: newLevel,
+		next:  make([]*Node, newLevel+1),
+	}
+
```

Comment:
The insert method does not check for duplicates. If duplicates are allowed, the implementation is fine.

Reply:
// Test duplicate insertions
	skipList.insert(30)
	skipList.insert(30)
	skipList.display()
This test case ensured that duplicate elements are inserted and displayed.
Output:
Skip List:
Level 9:
Level 8:
Level 7:
Level 6:
Level 5:
Level 4:
Level 3:
Level 2:
Level 1: 30
Level 0: 30 30

Reply:
Cool then